vector/src/main/java/im/vector/app/core/usecase/VectorBaseInOutUseCase.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.usecase
+
+interface VectorBaseInOutUseCase<T, R> {
```

Comment:
do you think we should enforce every use case to have its own request input type?

I'm wondering if we'll end up with multiple declarations (which rx/coroutines are also doing~)

```kotlin
VectorBaseInOutUseCase2<T, T2, R>
VectorBaseInOutUseCase3<T, T3, R>
// etc
```

Reply:
And I kind of think we shouldn't not have interface/base class for UseCase. It's not really solving anything, having just a naming convention would be enough in my opinion. Also UseCase don't absolutely have to be a class...

Reply:
As discussed earlier, I removed the interface.